src/main/java/seedu/address/logic/parser/AddPersonCommandParser.java

```
@@ -45,6 +45,14 @@ public AddPersonCommand parse(String args) throws ParseException {
         Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
+        // tag's limit is 30 characters, but we set it to 32 to account for the square brackets that encapsulate it
+        if (name.toString().isEmpty() || name.toString().length() > 100
+                || phone.toString().length() < 3 || phone.toString().length() > 20
+                || email.toString().length() < 6 || address.toString().length() > 200
+                || tagList.stream().anyMatch(tag -> tag.toString().length() > 32)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddPersonCommand.MESSAGE_USAGE));
+        }
+
```

Would it be better if you extract these checks into the constructor of each Object? 

That way, it is easier to maintain, plus we can throw a more detailed error message (eg: Address too long vs Invalid Command format)

---

@jiakai-17 i could do that. But would throwing a more detailed error message be counted as adding a new feature? I was under the impression that we could add limitations to the length of the fields so long as the error message was the same i.e. invalid command error msg

---

@freshcabbage123 Okay, I guess the feature freeze limits us from writing more descriptive error messages. Do also remember to add a point about more descriptive error messages in the Planned Enhancements section.